# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PWindowGlobal.ipdl
# Commit: 46af55564501
# Full Hash: 46af555645013a126416a440b658f2969eca4b20
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-06 08:52:03
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 55c4afa3a46c -r 46af55564501 dom/ipc/PWindowGlobal.ipdl
--- a/dom/ipc/PWindowGlobal.ipdl	Thu Jun 05 14:53:36 2025 +0000
+++ b/dom/ipc/PWindowGlobal.ipdl	Thu Jun 05 14:53:36 2025 +0000
@@ -15,10 +15,10 @@
 include protocol PInProcess;
 include protocol PBrowserBridge;
 include protocol PWebAuthnTransaction;
+include protocol PWebIdentity;
 
 include DOMTypes;
 include ClientIPCTypes;
-include IPCIdentityCredential;
 include NeckoChannelParams;
 include SessionStoreTypes;
 
@@ -67,6 +67,7 @@
 {
   manager PBrowser or PInProcess;
   manages PWebAuthnTransaction;
+  manages PWebIdentity;
 
 child:
   async __delete__();
@@ -203,21 +204,6 @@
 
   async ReloadWithHttpsOnlyException();
 
-  // Used by the Credential Manager API and FedCM to keep the discovery of
-  // a credential abstracted from the content process. This is required because
-  // credentialed requests that are specifically not partitioned are made and
-  // the results must not enter the child process until the user consents via
-  // purpose built UI.
-  async GetIdentityCredential(IdentityCredentialRequestOptions aOptions,
-                              CredentialMediationRequirement aMediationRequirement,
-                              bool aHasUserActivation)
-    returns (IPCIdentityCredential? identityCredential, nsresult rv);
-  async DisconnectIdentityCredential(IdentityCredentialDisconnectOptions aOptions) returns (nsresult rv);
-  async PreventSilentAccess() returns (nsresult rv);
-
-  // Indicate that the page has reported itself as logged in
-  async SetLoginStatus(LoginStatus foo) returns (nsresult rv);
-
   async GetStorageAccessPermission(bool aIncludeIdentityCredential) returns(uint32_t permission_action);
 
 
@@ -236,6 +222,7 @@
   async RecordUserActivationForBTP();
 
   async PWebAuthnTransaction();
+  async PWebIdentity();
 
 child:
   async NotifyPermissionChange(nsCString type, uint32_t permission);